neanderthal/src/main/kotlin/au/com/outware/neanderthal/data/repository/VariantSharedPreferencesRepository.kt

```diff
@@ -29,19 +31,32 @@ class VariantSharedPreferencesRepository(val klass: Class<out Any>,
         sharedPreferences = context.getSharedPreferences("${context.packageName}$SHARED_PREFERENCES_FILE_NAME", Context.MODE_PRIVATE)
         editor = sharedPreferences.edit()
 
+        val structure = klass.declaredFields.filter {
+            field -> !Modifier.isPrivate(field.modifiers) && !Modifier.isTransient(field.modifiers)
+        }.map {
+            field -> field.name
+        }.toHashSet()
+
         if (!sharedPreferences.contains(VARIANT_LIST)) {
-            editor.putStringSet(VARIANT_LIST, baseVariants.keys)
-            for(variant in baseVariants) {
-                editor.putString(variant.key, gson.toJson(variant.value))
-            }
-            editor.putString(CURRENT_VARIANT, defaultVariant)
-            editor.apply()
+            initialiseRepository(baseVariants, defaultVariant, structure)
+        } else if(!sharedPreferences.getStringSet(VARIANT_STRUCTURE, structure).equals(structure)) {
+            editor.clear()
+            initialiseRepository(baseVariants, defaultVariant, structure)
```

Comment:
Could you not just simplify this to
if(!sharedPreferences.getStringSet(VARIANT_STRUCTURE, structure).equals(structure)){
editor.clear();
}
initialiseRepository(baseVariants, defaultVariant, structure);


??